### PR TITLE
fix: navbar not showing login when user Logged Out

### PR DESCRIPTION
**Pull request by WYGIN**
### Description:

When the user is not logged in the website is showing similar screen with profile, similar to when the user is loggedin
### Dev Tested:
- [x] Yes
Picsart_23-07-17_18-28-38-213

#### After: 
Screenshot_20230526-163742

### CODE REVIEW

**Review by bhtibrewal — APPROVED** [review]

File: __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx [hunk 1]
```diff
@@ -16,9 +16,11 @@ describe('Progress page test', function () {
         );
         const textBoxes = screen.getAllByRole('textbox');
         expect(textBoxes).toHaveLength(3);
-        const button = screen.getByRole('button') as HTMLButtonElement;
-        expect(button).toBeInTheDocument();
-        expect(button.type).toBe('submit');
+        const buttons = screen.getAllByRole('button') as Array<HTMLButtonElement>;
```
Comment (shubhamsigdar1) [inline __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx:19]: Hey sai kiran @WYGIN  why you are changing progressUpdatePage unit test?
Can you please give some context??

**Review by shubhamsigdar1 — COMMENTED** [review]
Sai Kiran great work
I checked in my local 
and Its worked fine in my local but I Donot have much knowledge about useContext
So I have many doubt why we not used useState??

Please take review once by @bhtibrewal also(she has great knowledge in react)

**Comment by WYGIN** [comment]
Ok i will

**Review by shubhamsigdar1 — COMMENTED** [review]
Hi Sai Kiran, You provided in description that you are fixing 
image

But Now you start changing other files also...Is all these file solving problem of issue that you written in description??

If not please work on particular issue which have to solve and for other fixing make a Seperate issue ticket

**Comment by WYGIN** [comment]
> Hi Sai Kiran, You provided in description that you are fixing image
> 
> But Now you start changing other files also...Is all these file solving problem of issue that you written in description??
> 
> If not please work on particular issue which have to solve and for other fixing make a Seperate issue ticket

No, all those file changes are related to fixing the test issue & form should have a default role of form but in my case it is making my tests fail again so i made some changes to other files too.
Take a look at this url: https://html.spec.whatwg.org/multipage/forms.html#the-form-element

File: .gitpod.yml
Comment (RitikJaiswal75) [inline .gitpod.yml:1]: I don't think we need this file, Please remove it.

File: __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx [hunk 2]
```diff
@@ -16,7 +16,10 @@ describe('Progress page test', function () {
         );
         const textBoxes = screen.getAllByRole('textbox');
         expect(textBoxes).toHaveLength(3);
-        const button = screen.getByRole('button') as HTMLButtonElement;
+        const button = getByRole(
+            screen.getByRole('form') as HTMLFormElement,
+            'button'
+        ) as HTMLButtonElement;
```
Comment (RitikJaiswal75) [inline __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx:22]: why was this added? this is a part of progress updates, why should it be concerned with the navbar?

File: src/components/ProgressForm/ProgressForm.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ const ProgressForm = ({ questions }: formProps) => {
     };
 
     return (
-        <form className={styles.form}>
+        <form className={styles.form} role="form">
```
Comment (RitikJaiswal75) [inline src/components/ProgressForm/ProgressForm.tsx:76]: i don't think this was necessary.

**Review by RitikJaiswal75 — CHANGES_REQUESTED** [review]

File: src/components/ProgressForm/ProgressForm.tsx (hunk 3 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx]: did you check if the task progress page was working after these changes?

Comment (WYGIN) [reply src/components/ProgressForm/ProgressForm.tsx]: > did you check if the task progress page was working after these changes?


Is this the one your asking ?
i am unable to go to the route /progress/:id , do i need a special permission ?

File: __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx (hunk 1 above)
Comment (shubhamsigdar1) [reply __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx]: Thank you Sai Kiran for giving very good explanation

File: __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx (hunk 2 above)
Comment (shubhamsigdar1) [reply __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx:22]: So now you check only the button inside the form i.e. submit button...before it check all the button in progressUpdatePage?? Is I am right??

**Review by shubhamsigdar1 — COMMENTED** [review]
great work @WYGIN please reply my comment...I will approve the pr
https://github.com/Real-Dev-Squad/website-status/pull/589#discussion_r1211049701

File: __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx (hunk 2 above)
Comment (WYGIN) [reply __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx:22]: No, since the test is written targetting the progressUpdatePage's form element (because when the test is written there is no login button that render in progressUpdatePage) (AFAIK) checking that there is one button in one form & i dont want to break the test by ignoring it.
So i rewritten the test in the same way it used to work, so now it ignores NavBar & NavBar has seperate test https://github.com/Real-Dev-Squad/website-status/blob/develop/__tests__/Unit/Components/Navbar/Navbar.test.tsx

**So now this test only checks that there will be only one button in one form & there is only one form in one progressUpdatePage component**

**Review by shubhamsigdar1 — DISMISSED** [review]

File: src/components/ProgressForm/ProgressForm.tsx (hunk 3 above)
Comment (WYGIN) [reply src/components/ProgressForm/ProgressForm.tsx]: Screenshot_20230531-122001
Screenshot_20230531-121953
Screenshot_20230531-120924

**Review by bhtibrewal — DISMISSED** [review]
Looks okay to me

**Review by khushi818 — COMMENTED** [review]
LGTM

**Review by khushi818 — DISMISSED** [review]

**Review by sahsisunny — CHANGES_REQUESTED** [review]
Sign in button not showing yet

**Review by sahsisunny — DISMISSED** [review]
LGTM

**Comment by WYGIN** [comment]
@sahsisunny is the signin button working perfectly you mean or not? & thanks for the review

**Comment by sahsisunny** [comment]
> @sahsisunny is the signin button working perfectly you mean or not? & thanks for the review

It's working @WYGIN

**Review by RitikJaiswal75 — CHANGES_REQUESTED** [review]
Please attach test coverage to it

**Comment by WYGIN** [comment]
> Please attach test coverage to it

@RitikJaiswal75 added test coverage please check it

File: src/pages/challenges/index.tsx [hunk 4]
```diff
@@ -12,6 +12,12 @@ import { skipToken } from '@reduxjs/toolkit/dist/query';
 import useAuthenticated from '@/hooks/useAuthenticated';
 import { useGetChallengesQuery } from '@/app/services/challengesApi';
 
+export const loginNode = (
+    <a href={LOGIN_URL} target="_blank" rel="noreferrer">
+        Click here to Login
+    </a>
+);
+
```
Comment (bhtibrewal) [inline src/pages/challenges/index.tsx:20]: if this is being used in other files also, please take it out into a separate file and make it <LoginButton/>

File: __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx (hunk 2 above)
Comment (bhtibrewal) [reply __tests__/Unit/pages/Progress/ProgressUpdatesPage.test.tsx:22]: This is not a part of this Issue, please handle it separately

File: __mocks__/handlers/self.handler.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { rest } from 'msw';
+const URL = process.env.NEXT_PUBLIC_BASE_URL;
+
+export const userSelfData = {
```
Comment (bhtibrewal) [inline __mocks__/handlers/self.handler.ts:4]: please take a pull from develop, there should already be files having mock users data, creating a new one is not required IMO

File: __tests__/Unit/Components/Navbar/Navbar.test.tsx
Comment (RitikJaiswal75) [inline __tests__/Unit/Components/Navbar/Navbar.test.tsx:1]: I don't see any updates in tests that show the sign in button in the navbar

File: src/components/navBar/index.tsx [hunk 6]
```diff
@@ -76,7 +73,7 @@ const NavBar = () => {
                 </ul>
             </div>
             <div className={styles.userProfile}>
-                {!isLoggedIn ? (
+                {!isSuccess ? (
```
Comment (RitikJaiswal75) [inline src/components/navBar/index.tsx:76]: is success returned by rtk will only return if fetch is successful or not, it wont return proper response in case the data comes in blank, because the fetch was successful.

Comment (WYGIN) [reply src/components/navBar/index.tsx:76]: IMO the data will not come as an empty object because if the response in with in 200 range then only isSuccess value is truthy & in that case the user data exists in database & if the user data exists in database then the response is never null or empty, maybe the user has incomplete user details in this case

File: __tests__/Unit/Components/Navbar/Navbar.test.tsx [hunk 7]
```diff
@@ -91,4 +93,22 @@ describe('Navbar', () => {
...
+    test('show login in button when the user is not logged in', () => {
+        server.use(
+            selfHandlerFn(404, {
+                message: 'User does not exist',
+                error: 'Not Found',
+            })
+        );
+        renderWithProviders(<NavBar />);
+        expect(getSignInNode('button')).toBeVisible();
+    });
```
Comment (WYGIN) [inline __tests__/Unit/Components/Navbar/Navbar.test.tsx:106]: @RitikJaiswal75 this is the test that checks if the sign in button is showing or not, the tests should fail if the login button not showing

File: src/components/navBar/navBar.module.scss
Comment (sahsisunny) [inline src/components/navBar/navBar.module.scss:1]: Could you please create the variables for the color and specify the usage of REM instead of PX in case the value is less than 10PX?

**Comment by prakashchoudhary07** [comment]
Tests are failing, please look into it
